pkg/client/unversioned/thirdpartyresources.go

```diff
@@ -0,0 +1,102 @@
+	Create(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	Update(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	UpdateStatus(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	Delete(name string) error
+	Watch(opts unversioned.ListOptions) (watch.Interface, error)
+}
+
+// thirdPartyResources implements DaemonsSetsNamespacer interface
+type thirdPartyResources struct {
+	r  *ExtensionsClient
+	ns string
+}
+
+func newThirdPartyResources(c *ExtensionsClient, namespace string) *thirdPartyResources {
+	return &thirdPartyResources{c, namespace}
+}
+
+// Ensure statically that thirdPartyResources implements ThirdPartyResourcesInterface.
+var _ ThirdPartyResourceInterface = &thirdPartyResources{}
+
+func (c *thirdPartyResources) List(opts unversioned.ListOptions) (result *extensions.ThirdPartyResourceList, err error) {
+	result = &extensions.ThirdPartyResourceList{}
+	err = c.r.Get().Namespace(c.ns).Resource("thirdpartyresources").VersionedParams(&opts, api.Scheme).Do().Into(result)
+	return
+}
+
+// Get returns information about a particular daemon set.
+func (c *thirdPartyResources) Get(name string) (result *extensions.ThirdPartyResource, err error) {
+	result = &extensions.ThirdPartyResource{}
+	err = c.r.Get().Namespace(c.ns).Resource("thirdpartyresources").Name(name).Do().Into(result)
+	return
+}
+
+// Create creates a new daemon set.
+func (c *thirdPartyResources) Create(daemon *extensions.ThirdPartyResource) (result *extensions.ThirdPartyResource, err error) {
+	result = &extensions.ThirdPartyResource{}
+	err = c.r.Post().Namespace(c.ns).Resource("thirdpartyresources").Body(daemon).Do().Into(result)
+	return
+}
+
+// Update updates an existing daemon set.
+func (c *thirdPartyResources) Update(daemon *extensions.ThirdPartyResource) (result *extensions.ThirdPartyResource, err error) {
+	result = &extensions.ThirdPartyResource{}
+	err = c.r.Put().Namespace(c.ns).Resource("thirdpartyresources").Name(daemon.Name).Body(daemon).Do().Into(result)
+	return
+}
+
+// UpdateStatus updates an existing daemon set status
+func (c *thirdPartyResources) UpdateStatus(daemon *extensions.ThirdPartyResource) (result *extensions.ThirdPartyResource, err error) {
+	result = &extensions.ThirdPartyResource{}
+	err = c.r.Put().Namespace(c.ns).Resource("thirdpartyresources").Name(daemon.Name).SubResource("status").Body(daemon).Do().Into(result)
+	return
+}
+
+// Delete deletes an existing daemon set.
+func (c *thirdPartyResources) Delete(name string) error {
+	return c.r.Delete().Namespace(c.ns).Resource("thirdpartyresources").Name(name).Do().Error()
+}
+
+// Watch returns a watch.Interface that watches the requested daemon sets.
+func (c *thirdPartyResources) Watch(opts unversioned.ListOptions) (watch.Interface, error) {
+	return c.r.Get().
+		Prefix("watch").
+		Namespace(c.ns).
+		Resource("thirdpartyresources").
+		Param("resourceVersion", opts.ResourceVersion).
```

Comment:
I don't know if this got lost in the last round of comments, but this is not necessary. This will get added in VersionedParams: https://github.com/kubernetes/kubernetes/blob/master/pkg%2Fclient%2Funversioned%2Frequest.go#L425